pkg/controller/job/job_controller.go

```diff
@@ -1384,6 +1411,7 @@ func jobSuspended(job *batch.Job) bool {
 func (jm *Controller) manageJob(ctx context.Context, job *batch.Job, jobCtx *syncJobCtx) (int32, string, error) {
```

Comment:
IIUC, this function needs to be only adjusted regarding pod creation. Assume `podReplacementPolicy=Failed`, and:
```
active=4
terminating=3
wantActive=10
```
Then, we need to be creating pods when `active + terminating < wantActive`. So that we don't recreate `terminating` pods. I guess we can define the `diff := wantActive - active`. Then, `if podFailurePolicy=Failed {diff -= terminatingPodsCnt }`And enter the pod creation if `diff > 0`. Then, the list of `terminatingPods` is only needed for indexed jobs, so its computation can be delayed (IIUC).

Reply:
I'm a little unsure on this one.  I added an idea in manageJob for this.  

I believe that we want to the recreation behavior for both IndexedJobs and nonIndexedJobs.  

I tried to add this for both cases but I haven't got a chance to test it yet.